src/IECore/MemoryIndexedIO.cpp

```diff
@@ -36,106 +36,195 @@
 #include "IECore/FileIndexedIO.h"
 #include "IECore/VectorTypedData.h"
 
+#include <iosfwd>
+#include <boost/iostreams/stream.hpp>
+#include <boost/iostreams/positioning.hpp>
+
 using namespace IECore;
 
 IE_CORE_DEFINERUNTIMETYPEDDESCRIPTION( MemoryIndexedIO )
 
-///////////////////////////////////////////////
+//////////////////////////////////////////////////////
 //
-// FileIndexedIO::StreamFile (begin)
+// FileIndexedIO::StreamFile subclass implementations
 //
-///////////////////////////////////////////////
+//////////////////////////////////////////////////////
 
-class MemoryIndexedIO::StreamFile : public StreamIndexedIO::StreamFile
+class MemoryIndexedIO::CharVectorDataDevice
```

Comment:
No wait, I guess the benefit is that this way, if you only ever call `CharVectorDataDevice::read()`, we never call `writable()`, whereas we would have to initialise a CharVectorDevice with `myData->writable()`, and possibly pay for a copy?